tests/generator-Tests/Unit-Tests/CodeGenerationOptionsTests.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using MonoDroid.Generation;
+using NUnit.Framework;
+
+namespace generatortests
+{
+	[TestFixture]
+	public class CodeGenerationOptionsTests
+	{
+		[Test]
+		public void GetOutputNameUseGlobal ()
+		{
+			var opt = new CodeGenerationOptions { UseGlobal = true };
+
+			Assert.AreEqual (string.Empty, opt.GetOutputName (string.Empty));
+			Assert.AreEqual ("int", opt.GetOutputName ("int"));
+			Assert.AreEqual ("void", opt.GetOutputName ("void"));
+			Assert.AreEqual ("void", opt.GetOutputName ("System.Void"));
+			Assert.AreEqual ("params int[]", opt.GetOutputName ("params int[]"));
+			Assert.AreEqual ("params global::System.Object[]", opt.GetOutputName ("params System.Object[]"));
+
+			Assert.AreEqual ("global::System.Collections.Generic.List<string[]>",
+				opt.GetOutputName ("System.Collections.Generic.List<string[]>"));
+
+			Assert.AreEqual ("global::System.Collections.Generic.Dictionary<string, string>",
+				opt.GetOutputName ("System.Collections.Generic.Dictionary<string, string>"));
+
+			Assert.AreEqual ("global::System.Collections.Generic.List<global::System.Collections.Generic.List<string>>",
+				opt.GetOutputName ("System.Collections.Generic.List<System.Collections.Generic.List<string>>"));
+
+			Assert.AreEqual ("global::System.Collections.Generic.List<global::System.Collections.Generic.Dictionary<string, global::System.Collections.Generic.Dictionary<global::System.Object, global::System.Object>>>",
+				opt.GetOutputName ("System.Collections.Generic.List<System.Collections.Generic.Dictionary<string, System.Collections.Generic.Dictionary<System.Object, System.Object>>>"));
+
+			Assert.AreEqual ("global::System.Collections.Generic.IList<global::Kotlin.Pair>",
+				opt.GetOutputName ("System.Collections.Generic.IList<Kotlin.Pair>"));
+
+			Assert.AreEqual ("global::System.Collections.Generic.IDictionary<string, global::System.Collections.Generic.IList<global::Kotlin.Pair>>",
+				opt.GetOutputName ("System.Collections.Generic.IDictionary<string, System.Collections.Generic.IList<Kotlin.Pair>>"));
+
+			Assert.AreEqual ("global::System.Collections.Generic.IDictionary<global::System.Collections.Generic.IList<string>, global::Kotlin.Pair>",
+				opt.GetOutputName ("System.Collections.Generic.IDictionary<System.Collections.Generic.IList<string>, Kotlin.Pair>"));
+
+			Assert.AreEqual ("global::System.Collections.Generic.IDictionary<global::System.Collections.Generic.IList<string>, global::System.Collections.Generic.IList<global::Kotlin.Pair>>",
```

Comment:
These are all "positive" tests.

You should also test "negative" tests, "invalid" types.  What happens if it attempts to parse `System.Collections.Generic.List<>` (missing type) or `System.Collections.Generic.List<<>`  (too many `<`s) or `System.Collections.Generic.List<>>` (too many `>`s)?

Presumably it fails; *does* it?  Does it fail in the way we expect?  (Or does it instead return an empty string?)

Reply:
Should also try *nested* types, a'la

`List<List<string>.Enumerator[]>`.